05_stack_and_queue/src/sortStation.c

```diff
@@ -0,0 +1,119 @@
+#include "stack.h"
+
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int getPriority(char operation)
+{
+    if (operation == '*' || operation == '/') {
+        return 2;
+    } else if (operation == '+' || operation == '-') {
+        return 1;
+    }
+    return 0;
+}
+
+int sortStation(char* str, char* res)
+{
+    unsigned lastUsedResIndex = 0;
+    Stack* stack = newStack();
+    unsigned long len = strlen(str);
+    for (unsigned long i = 0; i < len; i++) {
+        if (isdigit(str[i])) {
+            if (lastUsedResIndex > 0 && res[lastUsedResIndex - 1] != ' ') {
+                res[lastUsedResIndex] = ' ';
+                lastUsedResIndex++;
+            }
+            while (i < len && isdigit(str[i])) {
+                res[lastUsedResIndex] = str[i];
+                lastUsedResIndex++;
+                i++;
+            }
+            if (i < len) {
+                i--;
+            }
```

Comment:
Есть у меня некоторые подозрения, что можно было написать вот так:

```suggestion
            res[lastUsedResIndex] = str[i];
            lastUsedResIndex++;
```

И поведение было бы такое же. Ну в смысле -- нет необходимости во вложенном цикле. Но это так -- с первого взгляда.

А то с этим сдвигом влево не совсем понятно выходит, что происходит.